### PR TITLE
Introduce SpanKind support

### PR DESCRIPTION
**Pull request by rakyll**
Updates #525.

### CODE REVIEW

File: trace/trace.go [hunk 1]
```diff
@@ -100,6 +100,13 @@ func WithSpan(parent context.Context, s *Span) context.Context {
 	return context.WithValue(parent, contextKey{}, s)
 }
 
+// All available span kinds.
+const (
+	SpanKindUnspecified = iota
```
Comment (odeke-em) [inline trace/trace.go]: Perfect, I was just about to comment on this, on the previous change to ask for this change :)

**Comment by rakyll** [comment]
@bogdandrutu @ramonza want to take a look before I merge? This is an API addition.

File: trace/trace.go [hunk 2]
```diff
@@ -111,9 +118,11 @@ type StartOptions struct {
 	// when there is a non-remote parent, no new sampling decision will be made:
 	// we will preserve the sampling of the parent.
 	Sampler Sampler
-}
 
-// TODO(jbd): Remove start options.
+	// SpanKind represents the kind of a span. If none is set,
+	// SpanKindUnspecified is used.
```
Comment (bogdandrutu) [inline trace/trace.go]: I would specify that these must be one of the SpanKindUnspecified, SpanKindServer, SpanKindClient

Comment (rakyll) [reply trace/trace.go]: Done

**Pull request merged**